File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.java
import java.io.IOException;
import java.util.Scanner;

public class Main {

	public static void main(String[] args) throws IOException {
		Scanner sc = new Scanner(System.in);

		int[] arr = new int[sc.nextInt()];
		for (int i = 0; i < arr.length; i++) {
			arr[i] = i + 1;
		}

		int N = sc.nextInt();
		for (int i = 0; i < N; i++) {
			int a = sc.nextInt() - 1;
			int b = sc.nextInt() - 1;
			
			while (a < b) {
				int temp = arr[a];
				arr[a++] = arr[b];
				arr[b--] = temp;
			}

		}
		String c = "";
		for (int j = 0; j < arr.length; j++) {
			c += arr[j] + " ";
		}
		System.out.print(c.trim());
		sc.close();
	}
}

File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.py
n, m = map(int,input().split())

arr = []
for i in range(n) :
    arr.append(i+1)

for _ in range(m) :
    i, j = map(int, input().split())
    
    i = i-1
    j = j-1
    
    while i < j :
        temp = arr[i]
        arr[i] = arr[j]
        arr[j] = temp
        i+= 1
        j -= 1
print(*arr)